src/reformatters/ecmwf/ecmwf_grib_index.py

```diff
@@ -0,0 +1,63 @@
+from collections.abc import Sequence
+from os import PathLike
+
+import pandas as pd
+
+from reformatters.common.config_models import DataVar
+from reformatters.ecmwf.ifs_ens.forecast_15_day_0_25_degree.template_config import (
+    EcmwfIfsEnsInternalAttrs,
+)
+
+
+def get_message_byte_ranges_from_index(
+    index_local_path: PathLike[str],
+    data_vars: Sequence[DataVar[EcmwfIfsEnsInternalAttrs]],
+    ensemble_member: int,
+) -> tuple[list[int], list[int]]:
+    """
+    Given an ECMWF grib index file, returns the byte ranges the given data var(s) & ensemble member can be found within.
+
+    Returns
+    -------
+    tuple[list[int], list[int]]
+        list of byte range starts & a list of byte range ends. Elements of each list in order of data vars.
+    """
+    byte_range_starts: list[int] = []
+    byte_range_ends: list[int] = []
+    index_file_df = _parse_index_file(index_local_path)
+    for data_var in data_vars:
+        rows = index_file_df.loc[
+            (ensemble_member, data_var.internal_attrs.grib_index_param),
+            ["_offset", "_length"],
+        ]
+        start, length = rows.values[0]
```

Comment:
lets assert there's only one matching row
```suggestion
        start, length = item(rows.values)
```